src/edge/edgeManager.ts

```diff
@@ -397,12 +398,25 @@ export class EdgeManager {
                 await Executor.executeCMD(outputChannel, "dotnet", { cwd: `${parent}`, shell: true }, `new aziotedgefunction -n "${name}" -r ${repositoryName}`);
                 break;
             case Constants.LANGUAGE_PYTHON:
-                const gitHubSource = "https://github.com/Azure/cookiecutter-azure-iot-edge-module";
-                const branch = Versions.pythonTemplateVersion();
-                await Executor.executeCMD(outputChannel,
-                    "cookiecutter",
-                    { cwd: `${parent}`, shell: true },
-                    `--no-input ${gitHubSource} module_name=${name} image_repository=${repositoryName} --checkout ${branch}`);
+                await new Promise((resolve, reject) => {
+                    tmp.dir({ unsafeCleanup: true }, (err, tmpDir, cleanupCallback) => {

```

Comment:
Why not just download to the module path? 
The benefit to use temp folder, is if error happens, no pollution to the project folder. If that is the intent, how about replace the repositoryname in the temp folder?

Reply:
There is no API to download a folder of a repo, so we use tmp folder.